send.c

```diff
@@ -82,7 +82,7 @@ int MPI_Isend(void *buf, int count, MPI_Datatype datatype,
       return(MPI_SUCCESS);
     }
 
-  sreq->buf=buf;
+  sreq->buf=(int*)buf;
```

Comment:
Can you confirm that this type cast is correct?

Reply:
Yes, sreq->buf is int* type in the code. Without this type cast, there is a confirmed compiler warning.